### PR TITLE
Front role collector

### PR DESCRIPTION
**Pull request by nicolasThal**
https://github.com/open-orchestra/open-orchestra-workflow-function-bundle/pull/77
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/1311

### CODE REVIEW

File: Backoffice/Collector/FrontRoleCollector.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace OpenOrchestra\Backoffice\Collector;
+
+use FOS\UserBundle\Model\UserInterface;
```
Comment (itkg-alavieille) [inline Backoffice/Collector/FrontRoleCollector.php]: use OpenOrchestra\UserBundle\Model\UserInterface no ?

File: Backoffice/Collector/FrontRoleCollector.php [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    {
+        $this->translator = $translator;
+    }
+
+    /**
+     * @return array
+     */
+    public function getRoles()
+    {
+        $this->roles[UserInterface::ROLE_DEFAULT] = $this->translator->trans(
```
Comment (itkg-alavieille) [inline Backoffice/Collector/FrontRoleCollector.php]: why add this role in this method and not in the constructor or in the compiler pass ?

File: Backoffice/Collector/RoleCollectorInterface.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+     */
+    public function addRole($role);
+
+    /**
+     * @param string $role
+     *
+     * @return bool
+     */
+    public function hasRole($role);
+}
```
Comment (itkg-alavieille) [inline Backoffice/Collector/RoleCollectorInterface.php]: eof

File: Backoffice/Collector/FrontRoleCollector.php (hunk 2 above)
Comment (nicolasThal) [inline Backoffice/Collector/FrontRoleCollector.php]: This way, I will only call the translator when I need the roles.

If I put it in the constructor, I will need the translator when I create the collector

**Comment by nicolasThal** [comment]
@itkg-alavieille pr updated

File: Backoffice/Collector/FrontRoleCollector.php (hunk 2 above)
Comment (itkg-alavieille) [inline Backoffice/Collector/FrontRoleCollector.php]: And why add a compiler pass to add this role like to backoffice role ?,

Comment (itkg-alavieille) [inline Backoffice/Collector/FrontRoleCollector.php]: But you add this roles each time than getRoles is call. why add a compiler pass to add this role like to backoffice role ?

**Comment by nicolasThal** [comment]
@itkg-alavieille I have followed your advice, the constructor is the best place to add the role I think

**Comment by itkg-alavieille** [comment]
:+1:

**Pull request merged**